src/transport_performance/gtfs/multi_validation.py

```diff
@@ -0,0 +1,476 @@
+"""Validating multiple GTFS at once."""
+from typing import Union
+from tqdm import tqdm
+import pathlib
+import glob
+import os
+
+from geopandas import GeoDataFrame
+import numpy as np
+import pandas as pd
+import folium
+from folium.plugins import FastMarkerCluster
+
+from transport_performance.gtfs.validation import GtfsInstance
+from transport_performance.utils.defence import (
+    _type_defence,
+    _is_expected_filetype,
+    _check_parent_dir_exists,
+    _enforce_file_extension,
+)
+
+
+class FileCountError(Exception):
+    """Raised when the number of files found is less than expected."""
+
+    pass
+
+
+class MultiGtfsInstance:
+    """Create a feed instance for multiple GTFS files.
+
+    This allows for multiple GTFS files to be cleaned, validated, summarised,
+    filtered and saved at the same time.
+
+    Parameters
+    ----------
+    path : Union[str, list]
+        A list of paths, or a glob string. See more informtion on glob strings
+        here: https://docs.python.org/3/library/glob.html
+
+    Attributes
+    ----------
+    paths : list
+        A list of the GTFS paths used to create the MultiGtfsInstance object.
+    instances : list
+        A list of GtfsInstance objects created from self.paths.
+    daily_trip_summary : pd.DataFrame
+        A combined summary of statistics for trips from all GTFS files in the
+        MultiGtfsInstance.
+    daily_route_summary : pd.DataFrame
+        A combined summary of statistics for routes from all GTFS files in the
+        MultiGtfsInstance.
+
+    Methods
+    -------
+    save()
+        Saves each GtfsInstance to a directory.
+    clean_feed()
+        Cleans all of the GTFS files.
+    is_valid()
+        Validates all of the GTFS files.
+    filter_to_date()
+        Filter all of the GTFS files to a specific date(s).
+    filter_to_bbox()
+        Filter all of the GTFS files to a specific bbox.
+    summarise_trips()
+        Create a summary of all of the routes throughout all GTFS files.
+    summarise_routes()
+        Create a summary of all of the trips throughout all GTFS files.
+
+    Raises
+    ------
+    TypeError
+        'path' is not of type string or list.
+    FileCountError
+        The glob string used for glob.glob does not find at least 2 files.
```

Comment:
This indicates that the new class won't work for single GTFS files, right? Could be that this class becomes the way to handle GTFS with detection of the number of GTFS files found controlling the logic used (eg 1 GTFS file use single GTFS, more than one use multi GTFS.)

Reply:
Done